openadmet/toolkit/filtering/filter_base.py

```diff
@@ -0,0 +1,142 @@
+    def set_mol_column(df: pd.DataFrame, smiles_column: str = "OPENADMET_CANONICAL_SMILES", mol_column: str = "mol") -> pd.DataFrame:
+        """
+        Set the mol column in the DataFrame if it does not exist.
+
+        Parameters
+        ----------
+        df : pandas.DataFrame
+            The input DataFrame to be filtered.
+        mol_column : str
+            The column name containing the RDKit Mol objects (default is 'mol').
+
+        Returns
+        -------
+        pandas.DataFrame
+            The DataFrame with the mol column set.
+        """
+
+        if mol_column in df.columns:
+            return df
+
+        if smiles_column not in df.columns:
+            raise ValueError(f"The DataFrame must contain a {smiles_column} column.")
+
+        df[mol_column] = df[smiles_column].apply(
+                lambda x: Chem.MolFromSmiles(x)
+            )
+        return df
+
+    @staticmethod
+    def min_max_filter(df: pd.DataFrame,
+                    property: str,
+                    min_threshold: float,
+                    max_threshold: float,
+                    mark_column: str,
+                    any_or_all: str = None) -> bool:
+        """
+        Filter a DataFrame based on a property value range.
+
+        Parameters
+        ----------
+        df : pandas.DataFrame
+            The input DataFrame containing the data to filter.
+        property : str
+            The name of the column in the DataFrame representing the property to filter on.
+        min_threshold : float
+            The minimum value of the property.
+        max_threshold : float
+            The maximum value of the property.
+        mark_column : str
+            The name of the column to store the boolean marks (True/False).
+        any_or_all : str, optional
+            If "any", the mark will be True if any of the conditions are met.
+            If "all", the mark will be True only if all conditions are met.
+            Default is pd.NA, which means no specific condition is applied.
+
+        Returns
+        -------
+        pandas.DataFrame
+            A DataFrame containing only rows where the property values are within the specified range.
+        """
+        if min_threshold is not None and max_threshold is None:
+            condition = df[property] >= min_threshold
+        elif max_threshold is not None and min_threshold is None:
+            condition = df[property] <= max_threshold
+        elif min_threshold is not None and max_threshold is not None:
+            condition = (df[property] >= min_threshold) & (df[property] <= max_threshold)
+        else:
+            raise ValueError("Either min_threshold or max_threshold must be provided.")
+
+        if not any_or_all:
+            df[mark_column] = condition
+        else:
+            if any_or_all == "any":
+                df[mark_column] = condition.any()
+            elif any_or_all == 'all':
+                df[mark_column] = condition.all()
+
```

Comment:
Catch invalid "Truthy" input e.g "hello"
```suggestion
            else:
                raise ValueError("invalid argument to any_or_all, must be one of 'any', 'all'")
```